zerver/lib/actions.py

```diff
@@ -4645,6 +4656,20 @@ def do_update_message(user_profile: UserProfile, message: Message,
         if new_stream is not None:
             assert stream_being_edited is not None
 
+            if subs_gaining_usermessages:
+                ums_to_create = []
+                for msg in changed_messages:
+                    for user_profile_id in subs_gaining_usermessages:
+                        # TODO: Add mention data to UserMessage.
```

Comment:
I think we're unlikely to want to do that TODO; perhaps this makes sense?

"""
The fact that the user didn't have a UserMessage originally means we can infer that the user
was not mentioned in the original message (even if mention syntax was present, it would not
take effect for a user who was not subscribed).  If we were editing the message's content, we
would rerender the message and then use the new stream's data to determine whether this is
a mention of a subscriber; but as we are not doing so, we choose to preserve the "was this 
mention syntax an actual mention" decision made during the original rendering for implementation
simplicity.  As a result, the only flag to consider applying here is `read`.
"""